File: forlab/ForLab.Web/ClientApp/src/@core/models/CMS/UsefulResource.ts
import { BaseEntityDto } from '../common/BaseEntityDto';
import { BaseFilter } from '../common/BaseFilter';

export class UsefulResourceDto extends BaseEntityDto {
  title: string = null;
  attachmentUrl: string = null;
  attachmentSize: number = null;
  attachmentName: string = null;
  extensionFormat: string = null;
  isExternalResource: boolean = false;
  downloadCount: number = null;
}

export class UsefulResourceFilterDto extends BaseFilter {
  title: string = null;
  attachmentUrl: string = null;
  attachmentSize: number = null;
  attachmentName: string = null;
  extensionFormat: string = null;
  isExternalResource: boolean = null;
  downloadCount: number = null;
}

File: forlab/ForLab.Data/DbModels/CMSSchema/UsefulResource.cs
using ForLab.Data.BaseModeling;
using System.ComponentModel.DataAnnotations.Schema;

namespace ForLab.Data.DbModels.CMSSchema
{
    [Table("UsefulResources", Schema = "CMS")]
    public class UsefulResource : BaseEntity
    {
        public string Title { get; set; }
        public string AttachmentUrl { get; set; }
        public float? AttachmentSize { get; set; }
        public string AttachmentName { get; set; }
        public string ExtensionFormat { get; set; }
        public bool IsExternalResource { get; set; } // false if the user broese and upload the resource itself
        public int DownloadCount { get; set; }
    }
}
